components/InterestZoneView/InterestZoneView.tsx

```diff
@@ -1,150 +1,88 @@
-            cat={cat}
-            isEditable={isEditable}
-          />
-        </AccordionDetails>
-      </Accordion>
-    </ListItem>
-  ));
-
-  return (
-    <div className={styles.container}>
-      <Box className={styles.box}>
-        <DetailsForm onChange={handleZoneInputChange} zone={props.zone} isEditable={isEditable} />
-        <div className={styles.catsDisplayDiv}>
-          <Accordion
-            className={styles.catTypeContent}
-            expanded={expanded === 'panel1'}
-            onChange={handleChange('panel1')}>
-            <AccordionSummary>
-              <Typography>Pisici sterilizate</Typography>
-            </AccordionSummary>
-            <AccordionDetails className={styles.catListContent}>
-              <div className={styles.catsDiv}>
-                <List className={styles.catsList}>{sterilizedCatsList}</List>
-              </div>
-            </AccordionDetails>
-          </Accordion>
-          <Accordion
-            className={styles.catTypeContent}
-            expanded={expanded === 'panel2'}
-            onChange={handleChange('panel2')}>
-            <AccordionSummary>
-              <Typography>Pisici nesterilizate</Typography>
-            </AccordionSummary>
-            <AccordionDetails className={styles.catListContent}>
-              <div className={styles.catsDiv}>
-                <List className={styles.catsList}>{unsterilizedCatsList}</List>
-              </div>
-            </AccordionDetails>
-          </Accordion>
-        </div>
-        <div className={styles.buttonsContainer}>
-          {!isEditable && (
-            <Button
-              variant='contained'
-              color='success'
-              className={styles.button}
-              onClick={handleEdit}>
-              Edit
-            </Button>
-          )}
-          {isEditable && (
-            <Button
-              variant='contained'
-              color='success'
-              className={styles.button}
-              onClick={handleSave}>
-              Save
-            </Button>
-          )}
-          <Button variant='outlined' color='error' className={styles.button} onClick={handleClose}>
-            Cancel
-          </Button>
-        </div>
-      </Box>
-    </div>
-  );
-};
+    const [isEditable, setIsEditable] = useState(false);
+    const [expanded, setExpanded] = useState<string | false>(false);
+
+    const handleChange = (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => {
+      setExpanded(isExpanded ? panel : false);
+    };
+    
+    useEffect(() => {
+        setIsEditable(false);  
+    }, [props.isVisible])
+
+
+    const handleEdit = () => {
+        setIsEditable(!isEditable);
+    };
+
+    const handleSave = () => {
+        setIsEditable(!isEditable);
+    };
+
+    const handleClose = useCallback(() => {
+        props.onClose();
+    }, []);
+
+    const handleSterilizedCatInputChange = () => {
```

Comment:
these functions are unused. either leave some TODOs explaining how they should be implemented, or remove them